File: src/com/sterndu/json/LongValue.java
package com.sterndu.json;

import java.util.function.Function;

public class LongValue implements NumberValue {

	private long value = 0;

	public LongValue(long value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public long getValueLong() { return value; }

	public void setValue(long value) { this.value = value; }

	@Override
	public String toJson() {
		return value + "";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return value + "";
	}

	@Override
	public JsonValue toJsonValue() {
		return this;
	}

	@Override
	public JsonValue toJsonValue(Function<Object, String> function) {
		return this;
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/LongValue.kt
@file:JvmName("LongValue")
package com.sterndu.json

class LongValue @JvmOverloads constructor(value: Long = 0) : NumberValue {
	var valueLong: Long = value
		private set

	override fun getValue(): Number {
		return valueLong
	}

	fun setValue(value: Long) {
		valueLong = value
	}

	override fun toJson(): String {
		return valueLong.toString() + ""
	}

	override fun toJson(function: (Any?) -> String): String {
		return valueLong.toString() + ""
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
